source/common/router/config_impl.h

```diff
@@ -86,6 +88,8 @@ class VirtualHostImpl : public VirtualHost {
   const std::string& name() const override { return name_; }
   const RateLimitPolicy& rateLimitPolicy() const override { return rate_limit_policy_; }
 
+  RequestHeaderParser& requestHeaderParser() const { return *request_headers_parser_; };
```

Comment:
ah, my bad... if this does not work could you try `return const RequestHeaderParser&`.
In general, you should not expose a reference to class internals in `const` method.